tests/main/auto-refresh-gating/task.yaml

```diff
@@ -0,0 +1,157 @@
+summary: Check that auto-refresh with gate-auto-refresh hooks works.
+
+details: |
+  Test auto-refresh with gate-auto-refresh hook support enabled
+  (experimental.gate-auto-refresh-hook feature) and verify the hook can control
+  automatic refreshes. The test uses two test snaps, one of them
+  being a content provider of the other. There are a few versions of these
+  snaps in the store (in stable/beta/edge channels) for this test.
+
+environment:
+  SNAP_NAME: test-snapd-refresh-control
+  CONTENT_SNAP_NAME: test-snapd-refresh-control-provider
+  CONTROL_FILE: /var/snap/test-snapd-refresh-control/common/control
+
+prepare: |
+  snap install --devmode jq
+  snap set system experimental.gate-auto-refresh-hook=true
+
+debug: |
+  jq -r '.data["snaps-hold"]' < /var/lib/snapd/state.json || true
+
+execute: |
+  force_autorefresh() {
+    echo "And force auto-refresh to happen"
+    jq ".data[\"last-refresh\"] = \"2007-08-22T09:30:44.449455783+01:00\"" /var/lib/snapd/state.json > /var/lib/snapd/state.json.new
+    mv /var/lib/snapd/state.json.new /var/lib/snapd/state.json
+  }
+
+  LAST_REFRESH_CHANGE_ID="0"
+
+  wait_for_autorefresh() {
+    local EXPECTED_SNAP="$1"
+    echo "wait for auto-refresh to happen"
+    for _ in $(seq 300); do
+      # get just the last auto-refresh change and exclude previous auto-refresh (if any) by its change id.
+      if CHANGES=$(snap changes | grep "Done.*Auto-refresh snap \"$EXPECTED_SNAP\"" | tail -1); then
+        local CHANGE_ID
+        CHANGE_ID=$(echo "$CHANGES" | awk '{print $1}')
+        if [ "$CHANGE_ID" -gt "$LAST_REFRESH_CHANGE_ID" ]; then
+          LAST_REFRESH_CHANGE_ID="$CHANGE_ID"
+          break
```

Comment:
perhaps here you could return and echo the value for LAST_REFRESH_CHANGE_ID, so then you can fail in case the change id didn't change, or at least to check if the refresh was completed

Reply:
Done.